projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.html

```diff
@@ -27,10 +27,24 @@
             [igxForItemSize]="renderedRowHeight"
             [igxForSizePropName]='getPropName(dim)'
             #verticalRowDimScrollContainer>
-        <igx-pivot-row-dimension-content  role='row' class="igx-grid-thead" [grid]="this"
-            [dimension]='dim'
+            <igx-pivot-row-dimension-content role='row' class="igx-grid-thead" [grid]="this"
+                [dimension]='dim'
+                [rowIndex]='rowIndex' [rowData]='rowData'
+                [density]="displayDensity" [width]="resolveRowDimensionWidth(dim)">
+            </igx-pivot-row-dimension-content>
+        </ng-template>
+    </div>
+    <div tabindex="0" *ngIf='rowDimensions.length === 0' #rowDimensionContainer role="rowgroup" class='igx-grid__tbody-pivot-dimension' (focus)="navigation.focusTbody($event)" (keydown)="navigation.handleNavigation($event)">
```

Comment:
I think this template can be simplified a bit. For example both can be extracted in their own templates so that it is more clear which is used when and they can be switched via ng-content, for example:

```
    <ng-content *ngTemplateOutlet="rowDimensions.length === 0 ? emptyRowDimensionsTemplate : defaultRowDimensionsTemplate"></ng-content>
```

Also the empty dimension does not need an igxForOf, it always renders just 1 record. It can be simplified to just something like this:

```
<ng-template #emptyRowDimensionsTemplate>
    <div tabindex="0" class='igx-grid__tbody-pivot-dimension' (focus)="navigation.focusTbody($event)">
        <igx-pivot-row-dimension-content role='row' class="igx-grid-thead" [grid]="this"
        [dimension]='emptyRowDimension'
        [rowIndex]='0' [rowData]='{}'
        [density]="displayDensity" [width]="resolveRowDimensionWidth(emptyRowDimension)">
    </igx-pivot-row-dimension-content>
    </div>
</ng-template>
```

Also passing undefined as dimension is not ideal. I would recommend implementing a default empty row dimension, which can store information like widths, since some features like resizing would not work with undefined.